daemon/api_notices.go

```diff
@@ -31,13 +32,16 @@ import (
 var noticeReadInterfaces = map[state.NoticeType][]string{
 	state.ChangeUpdateNotice:   {"snap-refresh-observe"},
 	state.RefreshInhibitNotice: {"snap-refresh-observe"},
+	state.SnapRunInhibitNotice: {"snap-refresh-observe"},
 }
 
 var (
 	noticesCmd = &Command{
-		Path:       "/v2/notices",
-		GET:        getNotices,
-		ReadAccess: interfaceOpenAccess{Interface: "snap-refresh-observe"},
+		Path:        "/v2/notices",
+		GET:         getNotices,
+		POST:        postNotices,
+		ReadAccess:  interfaceOpenAccess{Interface: "snap-refresh-observe"},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
```

Comment:
> I'm a bit unsure as to whether polkit authentication makes sense here

Good point regarding polkit, The polkit prompting doesn't make sense in this use case, I will remove it.

> if I understand correctly, these notices are issued by the snap client when a snap is run, to inhibit refreshes?

No, They are issued by the snap client to announce that **it is being inhibited from running** due to an ongoing refresh, so that a desktop client (specifically `snapd-desktop-integration` snap) can listen for this notice so it can display notifications or refresh progress as described in [SD167](https://docs.google.com/document/d/1HJQWKzgaB3yPKwRUqlxYhgyaKG5IcJe8eZOT4YY1CI8/edit#heading=h.g0v0i714voiv).

> Here, it checks whether the connection has the "snap-refresh-observe" interface connected, but the notice endpoint itself is on the usual snapd.socket, not snapd-snap.socket, and doesn't require a particular interface?

Thanks for pointing this out, It does look weird without context, It warrants a comment explaining it.
This marker interface check is just to be able do UX fallback:
- if no one is listening for the notices (i.e. `snap-refresh-observe` marker interface is not connected), snap run sends the notifications itself (https://github.com/snapcore/snapd/commit/63634cc73400e13c0dc6d286d8f62f2f4bae2246#diff-2040a7ab4147eafb94c9b58c40202df922bd298e0ec3d152e322cef32f051db5R115-R131).

We could send the notice regardless of the interface check (which actually might be better for readability and consistency) that's why it is not relevant to the API implementation, but I just chose the route with less API calls.

Reply:
I don't think we want authentication here as such

Reply:
Updated, Thank you!